### PR TITLE
Worked sign up and forgot password notifications

### PR DESCRIPTION
**Pull request by Ukhanskyi**
[Trello Task](https://trello.com/_____)

**CHECK LIST**
- [ ] specs added
- [x] specs passed
- [x] code coverage >90%
- [ ] DB indexes added if needed
- [ ] seed.rb file is updated if needed
- [x] PR is reviewed manually again (to make sure you have 100% ready code)
- [x] PR doesn't have conflicts with main branch
- [x] I've checked new feature as logged in and logged out user if needed
- [ ] "let!" is used in specs
- [x] screenshot of feature is added if task has UI

worked mailer notifications for sign up and for forgot password:

image

image

And add view for forgot password page and for change password change:

image

image

now for check sign up notification need fix ragistration page (not exist field for phone and sex)!!!

### CODE REVIEW

File: app/mailers/application_mailer.rb [hunk 1]
```diff
@@ -1,4 +1,4 @@
 class ApplicationMailer < ActionMailer::Base
-  default from: 'from@example.com'
+  default from: 'frankivsk.osbb@gmail.com'
```
Comment (vikdotdev) [inline app/mailers/application_mailer.rb:2]: оця штука б у конфігу сеталась, і десь з ENV

File: app/mailers/user_mailer.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class UserMailer < ApplicationMailer
+  default from: 'frankivsk.osbb@gmail.com'
```
Comment (vikdotdev) [inline app/mailers/user_mailer.rb:2]: і дублювати його тут не треба, наслідуванням мало б працювати

File: app/views/user_mailer/welcome_email.html.slim [hunk 3]
```diff
@@ -0,0 +1,19 @@
+html
+  head
+    meta content=("text/html; charset=UTF-8") http-equiv="Content-Type" /
+  body
+    div style='background-color: #cfdfe4; border-radius: 20px; width: 75%; margin: 0 auto;'
+      h1 style='border-radius: 20px; text-align: center; font-weight: bold; height: 60px; padding-top: 30px;'
```
Comment (vikdotdev) [inline app/views/user_mailer/welcome_email.html.slim:6]: лейаутів на мейли не маєте?

File: config/environments/development.rb [hunk 4]
```diff
@@ -59,4 +59,28 @@
...
+  config.action_mailer.smtp_settings = {
+    address: 'smtp.gmail.com',
+    port: 587,
+    domain: 'smtp.gmail.com',
+    user_name: ENV['MAILER_ADDRESS'],
+    password: ENV['MAILER_PASSWORD'],
+    authentication: 'plain',
+    enable_starttls_auto: true,
+    openssl_verify_mode: 'none'
+  }
```
Comment (vikdotdev) [inline config/environments/development.rb:85]: нащо у девелопмент режимі відправляти мейли?

File: app/views/user_mailer/welcome_email.html.slim (hunk 3 above)
Comment (Ukhanskyi) [reply app/views/user_mailer/welcome_email.html.slim:6]: виправив

File: app/mailers/user_mailer.rb (hunk 2 above)
Comment (Ukhanskyi) [reply app/mailers/user_mailer.rb:2]: виправив

File: app/mailers/application_mailer.rb (hunk 1 above)
Comment (Ukhanskyi) [reply app/mailers/application_mailer.rb:2]: Виправив

File: config/environments/development.rb (hunk 4 above)
Comment (Ukhanskyi) [reply config/environments/development.rb]: Done

File: app/views/devise/mailer/reset_password_instructions.html.slim [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+  hr style='width: 90%; text-align: center;'
+  p style='margin: 20px;'
+    p style='width: 90%; margin: 5px 5%; font-size: 20px;'
+      | Someone has requested a link to change your password. You can do this through the link below.
+    p style='width: 90%; margin: 5px 5%; font-size: 20px;'
+      = link_to 'Change my password', edit_password_url(@resource, reset_password_token: @token)
+    p style='width: 90%; margin: 5px 5%; font-size: 20px;'
+      | If you didn't request this, please ignore this email.
+    p style='width: 90%; margin: 5px 5%; font-size: 18px; padding-bottom: 30px'
+      | Your password won't change until you access the link above and create a new one.
```
Comment (mmarusyk) [inline app/views/devise/mailer/reset_password_instructions.html.slim:15]: Use styles in scss files

Comment (Ukhanskyi) [reply app/views/devise/mailer/reset_password_instructions.html.slim:15]: I'll be happy to use, but bootstrap and scss can't working

**Review by IvanDanyliv — APPROVED** [review]

File: app/javascript/stylesheets/application.scss [hunk 6]
```diff
@@ -30,6 +30,16 @@ body{
...
 }
 
+.btn-submit {
+  color: whitesmoke;
+  background-color: #798db6;
+}
+
+.btn-submit:hover {
+  color: whitesmoke;
+  background-color: #556588;
```
Comment (pereslop) [inline app/javascript/stylesheets/application.scss:40]: In 30 line is  the same property with the same value. Pleasebrrad about scss constants, and DRY i

File: app/models/user.rb [hunk 7]
```diff
@@ -28,6 +28,12 @@ class User < ApplicationRecord
 
   enum role: ROLES
 
+  after_create :welcome_email
+
+  def welcome_email
```
Comment (pereslop) [inline app/models/user.rb:33]: send_welcome_email, please read something about method naming conventions

File: app/views/user_mailer/welcome_email.html.slim [hunk 8]
```diff
@@ -0,0 +1,15 @@
+div style='background-color: #cfdfe4; border-radius: 20px; width: 75%; margin: 0 auto;'
+  h1 style='border-radius: 20px; text-align: center; font-weight: bold; height: 60px; padding-top: 30px;'
+    | Welcome to OSBB Application, #{@user.full_name}!
+  hr style='width: 90%; text-align: center;'
+  p style='margin: 20px;'
+    p style='width: 90%; margin: 5px 5%; font-size: 20px;'
+      | You have successfully signed up to frankivsk.osbb@gmail.com,
```
Comment (chu4) [inline app/views/user_mailer/welcome_email.html.slim:7]: you dont sign up to an email but a web page
change to a link to website

Comment (chu4) [reply app/views/user_mailer/welcome_email.html.slim:7]: Add I18n for emails

**Comment by chu4** [comment]
Відцентруй форми на паролі,
бо нижня частина зсунута в ліво дивно виглядає

File: app/javascript/stylesheets/application.scss (hunk 6 above)
Comment (Ukhanskyi) [reply app/javascript/stylesheets/application.scss]: fix it

File: app/models/user.rb (hunk 7 above)
Comment (Ukhanskyi) [reply app/models/user.rb]: done

File: app/views/user_mailer/welcome_email.html.slim (hunk 8 above)
Comment (Ukhanskyi) [reply app/views/user_mailer/welcome_email.html.slim]: fixed

**Comment by Ukhanskyi** [comment]
image

image

image

image

image

image

File: app/views/devise/mailer/reset_password_instructions.html.slim (hunk 5 above)
Comment (matsevich) [reply app/views/devise/mailer/reset_password_instructions.html.slim]: You right, only inline styles work for email html files

**Review by matsevich — APPROVED** [review]

**Review by chu4 — APPROVED** [review]

**Pull request merged**